backstory/src/demo/java/com/google/sps/servlets/PerspectiveServlet.java

```diff
@@ -39,11 +45,94 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.setContentType("application/json;");
     Gson gson = new Gson();
 
-    String[] cannedOutput = new String[0];
+    // will hold text to analyze
+    String text;
 
-    // write canned output to servlet
-    String json = gson.toJson(cannedOutput);
+    String json = request.getReader().readLine();
 
-    response.getWriter().println(json);
+    // get the text from the JSON & handle error if it cannot be converted
+    try {
+      JSONObject jsonContainer = new JSONObject(json);
```

Comment:
Nit: how about naming this `jsonObject` to better match its type? `jsonContainer` is not any more descriptive (IMO), but introduces uncertainty about the variable's type.

Reply:
That makes sense. I'll make the change.

Reply:
Done